modules/rest-compatibility/src/main/java/org/elasticsearch/rest/compat/version7/RestIndexActionV7.java

```diff
@@ -41,7 +42,7 @@
     static final String TYPES_DEPRECATION_MESSAGE = "[types removal] Specifying types in document "
         + "index requests is deprecated, use the typeless endpoints instead (/{index}/_doc/{id}, /{index}/_doc, "
         + "or /{index}/_create/{id}).";
-    private static final DeprecationLogger deprecationLogger = new DeprecationLogger(LogManager.getLogger(RestIndexAction.class));
+    private static final CompatibleApiLogger deprecationLogger = new CompatibleApiLogger(LogManager.getLogger(RestIndexAction.class));
```

Comment:
@jakelandis this is a very early draft, but do you think we should be logging compatible warnings instead or in addition to deprecation logs?

Reply:
In addition to. I see different purposes, the compatibility logs should provide a description of how/what was applied to achieve compatibility for example. "Your request [PUT foo/mytype/1] includes a type [mytype] which is ignored to maintain compatibility. " (and may later we can include helpful links to more information).